include/seqan3/core/algorithm/concept.hpp

```diff
@@ -60,10 +89,15 @@ class config_element_base;
 //!\}
 //!\cond
 template <typename config_t>
-SEQAN3_CONCEPT config_element_specialisation = std::semiregular<std::remove_reference_t<config_t>> &&
-requires (config_t c)
+SEQAN3_CONCEPT config_element_specialisation = requires
 {
-    { std::remove_reference_t<config_t>::id };
+    requires std::is_base_of_v<seqan3::pipeable_config_element<config_t>, config_t>;
+    requires std::semiregular<config_t>;
```

Comment:
That's the point. The concept says, that the type is identified as a valid config_element type. Now, when I write a generic function that is trying to create an instance of that type, because it says it is semiregular but the type is in fact a reference or const reference then I cannot create an instance of it in the code. So that's why I decided that it is better that the caller makes sure that the concept is invoked with the right type., which is with cv and reference qualifiers stripped.